src/main/java/org/gradle/profiler/CliGradleClient.java

```diff
@@ -44,23 +44,24 @@ public void loadToolingModel(List<String> tasks, List<String> gradleArgs, List<S
 
     @Override
     public void runTasks(List<String> tasks, List<String> gradleArgs, List<String> jvmArgs) {
-        String gradleOpts = jvmArgs.stream().map(arg -> '"' + arg + '"').collect(Collectors.joining(" "));
+        String gradleOpts = quoteJvmArguments(daemon, jvmArgs);
 
         List<String> commandLine = new ArrayList<>();
         gradleBuildConfiguration.addGradleCommand(commandLine);
         commandLine.addAll(gradleArgs);
         commandLine.addAll(tasks);
         commandLine.add("-Dorg.gradle.daemon=" + daemon);
-        if (daemon) {
-            commandLine.add("-Dorg.gradle.jvmargs=" + gradleOpts);
-        } else {
+        if (!daemon) {
```

Comment:
Unrelated question, why do we add empty `jvmargs` here?

Reply:
I think this is because we want to make sure we don't start a daemon after all. Though I am not sure why exactly, I left it in place just in case.

Reply:
wolfs/pass-jvmOpts-via-gradle-opts[](url)